src/main/java/org/springframework/data/jpa/repository/query/ParameterMetadataProvider.java

```diff
@@ -232,19 +237,19 @@ public Object prepare(Object value) {
 
 				switch (type) {
 					case STARTING_WITH:
-						return String.format("%s%%", value.toString());
```

Comment:
Please avoid changes unrelated to the actual improvement we're trying to fix.

Reply:
Should I create another issue in JIRA to fix creation of unnecessary objects?

Reply:
Unless you have detailed profiling numbers showing that that change has significant impact on the performance I don't think it's worth the effort.